### PR TITLE
WL-3818/9: JS-Escaping and non-existent sites

### PR DESCRIPTION
**Pull request by nicholaswilson100**

### CODE REVIEW

**Comment by buckett** [comment]
How about doing this in bootstrap.jsp as shouldn't pretty much everything be escaped? For example i18n strings should be be escaped as well.

**Comment by nicholaswilson100** [comment]
yup okay

**Comment by nicholaswilson100** [comment]
I don't really want to be putting Java in jsp though and we don't seem to 'do' tags...

**Comment by nicholaswilson100** [comment]
We do some

**Comment by buckett** [comment]
All I mean is that scattering escaping all over the code leads to things
slipping through. In this case everything that ends up in the JSP should be
escaped. Could do it in the .java file before putting anything in the
request.

On 12 February 2015 at 17:08, Nick Wilson notifications@github.com wrote:

> We do some
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ox-it/wl-feedback/pull/52#issuecomment-74109927.

## 

  Matthew Buckett, VLE Developer, IT Services, University of Oxford

**Comment by nicholaswilson100** [comment]
oh I see yup agreed

**Comment by nicholaswilson100** [comment]
Sorry I've pushed this 'WL-3818' Jira onto a branch here called 'WL-3819' which is another but related Jira.

**Comment by nicholaswilson100** [comment]
If it's okay, I'm going to put Jira WL-3819 (non-existent sites) on this pull request too as it's dealing with the same files.

**Comment by nicholaswilson100** [comment]
(WL-3819 is) linked to https://github.com/ox-it/wl-tetraelf-hierarchy/pull/20

File: src/java/org/sakaiproject/feedback/util/SakaiProxy.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import java.text.MessageFormat;
 
 import org.apache.commons.fileupload.FileItem;
+import org.apache.commons.lang.StringEscapeUtils;
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/util/SakaiProxy.java:7]: Unused.

File: src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java [hunk 2]
```diff
@@ -114,7 +115,7 @@ private String handleReport(final EntityView view, final Map<String, Object> par
             return BAD_REQUEST;
         }
 
-        final String siteId = view.getPathSegment(1);
+        final String siteId = view.getPathSegment(1).replaceAll(FeedbackTool.FORWARD_SLASH, "/");
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/tool/entityproviders/FeedbackEntityProvider.java:118]: This sort of hack should have a note about why it's needed.

File: src/java/org/sakaiproject/feedback/tool/FeedbackTool.java [hunk 3]
```diff
@@ -140,12 +143,23 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
...
 
         response.setContentType("text/html");
         request.getRequestDispatcher("/WEB-INF/bootstrap.jsp").include(request, response);
     }
 
+	private void setStringAttribute(HttpServletRequest request, String key, String value){
+		request.setAttribute(key, StringEscapeUtils.escapeJavaScript(value));
+	}
+
+	private void setMapAttribute(HttpServletRequest request, String key, Map map){
```
Comment (buckett) [inline src/java/org/sakaiproject/feedback/tool/FeedbackTool.java:156]: It's probably worth keeping the generics on the Map, so you can never get a unexpected map that contains nested maps.

**Pull request merged**